File: app/src/main/java/com/wys/commonbaselib/jetpack/java/MyActivity.java
package com.wys.commonbaselib.jetpack.java;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import com.wys.commonbaselib.R;

public class MyActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_my);

        MyViewModel model = new ViewModelProvider(this).get(MyViewModel.class);
        model.getUsers().observe(this,users -> {
            //update UI
        });
    }
}

File: jetpack/src/main/java/com/aixuexi/jetpack/viewmodel/MyActivity.kt
package com.aixuexi.jetpack.viewmodel

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.util.Log
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.aixuexi.jetpack.R

class MyActivity : AppCompatActivity() {
    private val TAG = "MyActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_my)

        val model: MyViewModel  = ViewModelProvider(this).get(MyViewModel::class.java)
        model.getUsers().observe(this, Observer {
            users -> Log.d(TAG,"update UI")
            val d = users
        })

        Handler().postDelayed(Runnable {
            model.updateUser()
        },1000)
    }
}